File: lua/top_ten_scorers.lua
players = {}

for line in io.lines() do
  team, name, games, points = line:match("([^,]+),([^,]+),(%d+),(%d+)")
  if tonumber(games) >= 15 then
    table.insert(players, {name=name, team=team, ppg=points/games})
  end
end

table.sort(players, function (p1, p2) return p1.ppg > p2.ppg end)

for index, player in ipairs(players) do
  if index > 10 then 
    break
  end
  format = "%-22s%-4s%8.3f"
  print(string.format(format, player.name, player.team, player.ppg))
end

File: go/top_ten_scorers.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"sort"
	"strconv"
)

type Player struct {
	name string
	team string
	ppg  float64
}

func main() {
	players := make([]Player, 0)
	scanner := bufio.NewScanner(os.Stdin)
	re := regexp.MustCompile(`([^,]+),([^,]+),(\d+),(\d+)`)
	for scanner.Scan() {
		match := re.FindStringSubmatch(scanner.Text())
		games, _ := strconv.Atoi(match[3])
		points, _ := strconv.Atoi(match[4])
		if games >= 15 {
			ppg := float64(points) / float64(games)
			player := Player{match[2], match[1], ppg}
			players = append(players, player)
		}
	}
	sort.Slice(players, func(i, j int) bool {
		return players[i].ppg > players[j].ppg
	})
	for _, player := range players[:10] {
		format := "%-22s%-4s%8.2f\n"
		fmt.Printf(format, player.name, player.team, player.ppg)
	}
}
